Automattic/wp-calypso
# Post- Purchase: Set a migration as `pending-DIFM/DIF` when user arrives on the credential/instructions step

Related to  #96107 

## Proposed Changes
* Set migration as `migration-pending-*`  when the user arrives on a credential step or the instructions step. 
* Minor improvements on hooks that are updating the migration status. 

## Why are these changes being made?

## Testing Instructions

Scenario: DIFM
* Enable the feature flag  (DevTools> Console >  `sessionStorage.setItem('flags', 'automated-migration/pending-status'`);
* Go to any migration flow (E.g /setup/hosted-site-migration)
* Follow the steps, choosing the DIFM option until you reach the credential step
* Get the siteId on the URL (E.g `/ `/site-migration-credentials?siteId=XXXXX`)
* Check the sticker `migration-pending-difm` was applied using BlogRC  (Ping me if you have doubts about how to do it)

Scenario: DIF
* Enable the feature flag  (DevTools> Console >  `sessionStorage.setItem('flags', 'automated-migration/pending-status'`);
* Go to any migration flow (E.g /setup/hosted-site-migration)
* Follow the steps, choosing the DIY option until you reach the instructions page
* Get the siteId on the URL (E.g `/ `/site-migration-credentials?siteId=XXXXX`)
* Check the sticker `migration-pending-dif` was applied using BlogRC  (Ping me if you have doubts about how to do it)

> [!NOTE]  
> The current solution does not prevent the override of a migration-started-diy` with a `migration-pending-*`  it will be handled by https://github.com/Automattic/wp-calypso/issues/96114






## Pre-merge Checklist
